From stern@rowland.harvard.edu Thu Jun 22 10:28:13 2006
Date: Thu, 22 Jun 2006 13:28:11 -0400 (EDT)
From: Alan Stern <stern@rowland.harvard.edu>
To: Greg KH <greg@kroah.com>
cc: USB development list <linux-usb-devel@lists.sourceforge.net>, 
    Zoran Marceta <Zoran.Marceta@micronasnit.com>
Subject: usbfs: use the correct signal number for disconnection
Message-ID: <Pine.LNX.4.44L0.0606221325140.8819-100000@iolanthe.rowland.org>

From: Zoran Marceta <Zoran.Marceta@micronasnit.com>


usbfs stores the wrong signal number in the siginfo structure used for
notifying user programs about device disconnect.  This patch (as726)
fixes it.

From: Zoran Marceta <Zoran.Marceta@micronasnit.com>
Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/core/inode.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/core/inode.c
+++ gregkh-2.6/drivers/usb/core/inode.c
@@ -696,7 +696,7 @@ static void usbfs_remove_device(struct u
 		wake_up_all(&ds->wait);
 		list_del_init(&ds->list);
 		if (ds->discsignr) {
-			sinfo.si_signo = SIGPIPE;
+			sinfo.si_signo = ds->discsignr;
 			sinfo.si_errno = EPIPE;
 			sinfo.si_code = SI_ASYNCIO;
 			sinfo.si_addr = ds->disccontext;
